[DOCS-12615] Data Observability reorg (#33006)
DataDog/documentation: content/en/data_observability/integrations/openlineage/_index.md

---
title: OpenLineage
description: "Connect OpenLineage to Datadog to capture lineage from custom systems and orchestrators."
further_reading:
  - link: '/data_observability/'
    tag: 'Documentation'
    text: 'Data Observability'
  - link: '/data_observability/integrations/openlineage/datadog_agent_for_openlineage'
    tag: 'Documentation'
    text: 'Set up Datadog Agent for OpenLineage Proxy'
---

## Overview

Lineage in Datadog Data Observability is built on [OpenLineage][1], the open source standard for collecting and analyzing data lineage. OpenLineage is especially helpful when you need to:

- Capture lineage from systems Datadog doesn't integrate with natively, such as in-house tools
- Emit lineage events for jobs or orchestrators where building a native Datadog integration would be difficult

> Replace the hostname in the examples with the relevant [Datadog site][2] for your organization. You can find your Datadog site by [following these instructions][3]. This example uses `datadoghq.com`.

Reviewer: ```suggestion
Replace the hostname in the examples with the relevant [Datadog site][2] for your organization. To find your Datadog site, see [Access the Datadog site][3]. This example uses `datadoghq.com`.
```
Again, not a huge thing by any means, there's just a lot of links floating around and it can help to give some context about whether one stays in the docs or goes somewhere else!

    eventTime=datetime.utcnow().isoformat(),
    run=Run(runId=str(uuid.uuid4())),
    job=Job(namespace="default", name="test-job"),
    producer="your-producer-id"
)

client.emit(event)
```

## Option 3: OpenLineage Python client (Datadog transport)

In OpenLineage 1.37.0+, use the [Datadog transport][7] for automatic configuration and optimized event delivery.

```python
from datetime import datetime
import uuid
from openlineage.client import OpenLineageClient
from openlineage.client.event_v2 import RunEvent, RunState, Job, Run
from openlineage.client.transport.datadog import DatadogConfig, DatadogTransport

config = DatadogConfig(
    apiKey="your-datadog-api-key",
    site="datadoghq.com"  # Change if using a different Datadog site
)

client = OpenLineageClient(transport=DatadogTransport(config))

event = RunEvent(
    eventType=RunState.START,
    eventTime=datetime.utcnow().isoformat(),
    run=Run(runId=str(uuid.uuid4())),
    job=Job(namespace="default", name="test-job"),
    producer="your-producer-id"
)

client.emit(event)
```

<div class="alert alert-info">For Option 3, you can skip <code>DatadogConfig</code> by using environment variables:

> ```shell

Reviewer: Ooo, never seen a code sample in a note before, but it looks great!
